samcli/commands/local/lib/local_lambda.py

```diff
@@ -86,6 +88,33 @@ def is_debugging(self):
         """
         return bool(self.debug_context)
 
+    def _sanitize_inlinecode(self, inlinecode):
+        """
+        Verify if InlineCode needs runtime substitution
+        """
+        # InlineCode isn't type safe
+        if isinstance(inlinecode, dict):
+            # We expect to have only 1 keypair..
+            inlinecode = list(inlinecode.values())[0]
+        # If ${something} found inside the code, raise a NotImplemented error
+        if re.search(r'\${[^!]\S+}', inlinecode):
+            LOG.debug('Not able to parse inline code: %s', inlinecode)
+            raise NotImplementedError('Not able to parse inline code because of !Sub presence')
+        return inlinecode
+
+    # FIXME: randommize dump path and make sure not to overwrite an existing file
```

Comment:
Does it need to be randomized? Capturing why would be a good addition to the design doc above.

Reply:
The point of randomizing it is in case we're using the local directory to dump the file (instead tempdir) with the risk of overwriting existing files.. another issue i might see is having multiple Functions dumped for the same session (rare but i had the feeling could be a use case for that)

Reply:
If using logical ids to store inline code files and placing them in ***cache***, I think there's no need to..